integrations/clickup/clickup/client.py

```diff
@@ -0,0 +1,147 @@
+
+    async def _create_team_webhook_events(self, app_host: str, team_id: int) -> None:
+        webhook_target_app_host = f"{app_host}/integration/webhook"
+        webhooks_response = await self.client.get(
+            f"{self.clickup_url}/team/{team_id}/webhook"
+        )
+        webhooks_response.raise_for_status()
+        webhooks = webhooks_response.json()
+
+        existing_webhook = next(
+            (
+                webhook
+                for webhook in webhooks["webhooks"]
+                if webhook["endpoint"] == webhook_target_app_host
+            ),
+            None,
+        )
+
+        if existing_webhook:
+            logger.info(
+                f"Ocean real time reporting clickup webhook already exists [ID: {existing_webhook['id']}]"
+            )
+            return
+
+        webhook_create_response = await self.client.post(
+            f"{self.clickup_url}/team/{team_id}/webhook",
+            json={
+                "endpoint": webhook_target_app_host,
+                "events": WEBHOOK_EVENTS,
+            },
+        )
+        webhook_create_response.raise_for_status()
+        webhook_create = webhook_create_response.json()
+        logger.info(
+            f"Ocean real time reporting clickup webhook created "
+            f"[ID: {webhook_create['id']}, Team ID: {team_id}]"
+        )
+
+    async def get_teams(self, params: dict[str, Any] = {}) -> list[dict[str, Any]]:
+        teams_response = await self.client.get(
+            f"{self.clickup_url}/team", params=params
+        )
+        teams_response.raise_for_status()
+        teams = teams_response.json()["teams"]
+        return teams
+
+    async def _get_spaces(
+        self, team_id: str, params: dict[str, Any] = {}
+    ) -> list[dict[str, Any]]:
+        spaces_response = await self.client.get(
+            f"{self.clickup_url}/team/{team_id}/space", params=params
+        )
+        spaces_response.raise_for_status()
+        return spaces_response.json()["spaces"]
+
+    async def get_projects(self, params: dict[str, Any] = {}) -> list[dict[str, Any]]:
+        # getting all teams so as to retrieve the spaces within each team, then using their
+        # ids to get the projects (lists) within each space
+        projects: list[dict[str, Any]] = []
+        for team in await self.get_teams(params):
+            spaces = await self._get_spaces(team["id"], params)
+
+            for space in spaces:
+                projects_response = await self.client.get(
+                    f"{self.clickup_url}/space/{space['id']}/list", params=params
+                )
+                projects_response.raise_for_status()
+                # because the port-app-config uses the team ID to relate the projects to the teams
+                # we add the team ID to each project
+                # also, the clickup returns the datetime as a timestamp, so we convert it to a
+                # datetime object
```

Comment:
we thrive to write a generic code, as although we do have in thought how the mapping will be used, we should be aware that users can change it to their needs and therefor make it generic and as extensible as we can. 

e.g. instead of returning only the team_id, return the full team object.
also no need to mention the datetime and the transformation if you are not doing anything with it here